pkg/cmd/roachtest/test_runner.go

```diff
@@ -1273,6 +1297,11 @@ func (r *testRunner) runTest(
 		// NB: We're adding the timeout failure intentionally without cancelling the context
 		// to capture as much state as possible during artifact collection.
 		t.addFailure(0, "test timed out (%s)", timeout)
+		// If the Side-Eye integration was configured, capture a snapshot of the
+		// cluster before we destroy it.
+		if r.sideEyeClient != nil {
```

Comment:
Any reason this is not inside `teardownTest`? Logically, all post-processing happens there.

Reply:
In fact, `collectArtifacts` is even more fitting.